routing/payment_lifecycle.go

```diff
@@ -50,7 +50,72 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 		// If this payment had no existing payment attempt, we create
 		// and send one now.
 		if p.attempt == nil {
-			firstHop, htlcAdd, err := p.createNewPaymentAttempt()
+			// Before we attempt this next payment, we'll check to see if either
```

Comment:
This method is already rather massive, to not make things worse, how about wrapping this new logic in a layer of indirection (new method)?

Reply:
Done in later commits.